pkg/azuredx/azureauth/auth_test.go

```diff
@@ -87,6 +87,34 @@ func TestOnBehalfOf(t *testing.T) {
 	}
 }
 
+// TODO test feature flag disabled too -> should have same result
+func TestOnBehalfOfDisabled(t *testing.T) {
```

Comment:
elaborated in https://github.com/grafana/azure-data-explorer-datasource/issues/359#issuecomment-1136964347 but I don't think we _need_ the backend changes.

Reply:
I'll remove the TODO but will leave the test as currently it checks that if the feature is disabled in the data source it will not use OBO (i.e. use TokenProvider, not AADClient), we did not have a test for this 😉